### PR TITLE
Merge feeds calendar dates skip

### PR DESCRIPTION
**Pull request by landonreed**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JavaDoc and code is thoroughly commented
- [x] The description lists all applicable issues this PR seeks to resolve
- [x] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

- Address part 2 of #256 by adding calendar_dates skipping logic.
- ~WAITING ON: MTC to indicate whether they are ok with implementing changes to address item 1 of #256 (odd skipping logic).~ Confirmed, MTC is OK with this change.

### CODE REVIEW

File: src/main/java/com/conveyal/datatools/manager/jobs/MergeFeedsJob.java [hunk 1]
```diff
@@ -368,6 +368,14 @@ private int constructMergedTable(Table table, List<FeedToMerge> feedsToMerge,
                 Field[] fieldsFoundInZip =
                     table.getFieldsFromFieldHeaders(csvReader.getHeaders(), null);
                 List<Field> fieldsFoundList = Arrays.asList(fieldsFoundInZip);
+                // Initialize future feed's first date to the first calendar date from the validation result.
+                // This is equivalent to either the earliest date of service defined for a calendar_date record or the
+                // earliest start_date value for a calendars.txt record. For MTC, however, they require that GTFS
+                // providers use calendars.txt entries and prefer that this value (which is used to determine cutoff
+                // dates for the active feed when merging with the future) be strictly assigned the earliest
+                // calendar#start_date (unless that table for some reason does not exist).
+                LocalDate futureFeedFirstDate = feedsToMerge.get(0).version.validationResult.firstCalendarDate;
```
Comment (binh-dam-ibigroup) [inline src/main/java/com/conveyal/datatools/manager/jobs/MergeFeedsJob.java]: Is it feedsToMerge(0) or is it feedsToMerge(feedIndex)?

Comment (landonreed) [reply src/main/java/com/conveyal/datatools/manager/jobs/MergeFeedsJob.java]: It is `feedsToMerge(0)`, but to your point this should probably be moved out of the for loop block.

**Comment by evansiroky** [comment]
Since this is potentially omitting service_ids that might have a corresponding trip, shouldn't there be a check in the trips table for whether the service_id still exists? And if it doesn't the trip should not be added.

**Pull request merged**